tiledb/sm/consolidator/commits_consolidator.cc

```diff
@@ -98,19 +99,52 @@ Status CommitsConsolidator::consolidate(
       to_consolidate.front(), to_consolidate.back(), write_version);
   RETURN_NOT_OK(st1);
 
+  // Compute size of consolidated file. Save the sizes of the files to re-use
+  // below.
+  storage_size_t total_size = 0;
+  const auto base_uri_size = array_dir.uri().to_string().size();
+  std::vector<storage_size_t> file_sizes(to_consolidate.size());
+  for (uint64_t i = 0; i < to_consolidate.size(); i++) {
+    const auto& uri = to_consolidate[i];
+    total_size += uri.to_string().size() - base_uri_size + 1;
+
+    // If the file is a delete, add the file size to the count and the size of
+    // the size variable.
+    if (stdx::string::ends_with(
+            uri.to_string(), constants::delete_file_suffix)) {
+      RETURN_NOT_OK(storage_manager_->vfs()->file_size(uri, &file_sizes[i]));
```

Comment:
Should we be worried about the `vfs::file_size` calls here? I'm thinking this might translate into lots of sequential remote requests.

Reply:
Yes and no. Consolidation is not frequent, but if we can improve this moving forward, why not. I'll add an item on my list to follow up on this.